src/main/java/com/google/api/gax/grpc/RpcFuture.java

```diff
@@ -42,4 +42,11 @@
   void addListener(Runnable listener, Executor executor);
 
   void addCallback(RpcFutureCallback<? super V> callback);
+
+  <X extends Throwable> RpcFuture catching(
+      Class<X> exceptionType, Function<? super X, ? extends V> callback);
+
+  interface Function<F, T> {
```

Comment:
So I realized that we should probably put the new RpcFuture family of classes in com.google.api.gax.core instead of ...grpc, because it's not really tied to grpc. I do want this interface moved to its own file to mirror Guava. The class structure that users is more important to me than the file structure we see, and inner classes are more complex to users.